charts/topolvm/templates/node/daemonset.yaml

```diff
@@ -37,29 +37,73 @@ spec:
       {{- end }}
       serviceAccountName: {{ template "topolvm.fullname" . }}-node
       {{- if .Values.node.lvmdEmbedded }}
-      hostPID: true
+      # hostPID: true
       {{- end }}
       {{- with .Values.node.initContainers }}
       initContainers: {{ toYaml . | nindent 6 }}
       {{- end }}
       containers:
-        - name: topolvm-node
+      {{- if .Values.node.lvmdEmbedded }}
```

Comment:
lvmdEmbedded is fundamentally different from what you are trying to do here. Embedded means embedded in topolvm-node, not a separate binary. If you run lvmd separately from topolvm-node it is no longer embedded.

Reply:
Yes I do agree I changed the implementation and re-used the same flag the admin experience should be the same though would you like a new word here or a more verbose description of the change

Reply:
Just trying to add some context as the original implementor of the embedded functionality: IMO this is a fundamentally different deployment from using `--embed-lvmd` in topolvm-node so we should find a different name for it. 

From what I can see, this functionality should not be removed just because we change the way that we move from 2 DS to 1 DS without using this flag. When using the flag, we do not even need to run LVMD so there will always be 1 DS.

I have 2 follow-up questions from your work:
1. How in the future would we allow users to run with `--embed-lvmd` on topolvm-node? IMO this is answered by running 1 DS with 2 containers in the default setting (lvmdEmbedded = false) or introducing a new flag that is mutually exclusive with lvmdEmbedded =  true
2. What tangible benefit does it bring to have a single daemon set with 2 containers over 2 ds with 1 container. I currently do not see major gains, would be cool to have this in the PR explanation

Reply:
1) I see we have multiple implementation choices right now LVMD can be systemd, in its own DS or as a child process of the node DS container. In this PR I removed "embeded" as it is currently defined because that didn't seem to solve a unique problem or offer any value on its own and to me made troubleshooting more complicated. having LVMD as its own daemon set also doesn't seem to be particularly useful as an option because we could have race conditions between the pods or node may not start for some reason such as lack of resources. So really I moved the LVMD "managed" into this pod and redefined embeded. I think adding node.lvmd.external would be more descriptive as this really just reduces the deployment choices to two. DS (via node pod) and external

Reply:
> I see we have multiple implementation choices right now LVMD can be systemd, in its own DS or as a child process of the node DS container. 

It is not a child process if you run it within topolvm-node. It is one process and we do not open up a gRPC server in this case.

> In this PR I removed "embeded" as it is currently defined because that didn't seem to solve a unique problem or offer any value on its own and to me made troubleshooting more complicated. 

I believe you are quite quick to come to this resolution. Please reference the [dedicated design proposal](https://github.com/topolvm/topolvm/blob/main/docs/proposals/lvmd-within-topolvm-node.md) and see if you still do not understand the use case. Just because this option of deployment is not applicable for you does not mean it is not useful for other users.

> having LVMD as its own daemon set also doesn't seem to be particularly useful as an option because we could have race conditions between the pods or node may not start for some reason such as lack of resources. 

This does not make sense because 1. race conditions between pods are not a thing (we are an eventually consistent system, and starting one before the other does not result in problems) and 2. If the node does not have enough resources for both lvmd and topolvm-node and you run with 1DS the daemonset will still not start and run successfully.

> So really I moved the LVMD "managed" into this pod and redefined embeded. I think adding node.lvmd.external would be more descriptive as this really just reduces the deployment choices to two. DS (via node pod) and external

I disagree that you redefined embedded. You removed the option to configure it entirely. There is no way in your chart to enable this kind of configuration.
However I agree that having 2 deployment choices should be the goal. 
My conclusion is different however: I think your way of deploying (1 DS with 2 containers ) should succeed the 2 DS approach and NOT embedded LVMD.

Reply:
re race conditions I have seen instances of the node pod crashing when lvmd is not ready, I have also seen a few instances where workload pods start before node/lvmd is ready

Thanks for the doc pointer, after reading I think my approach achieves the same goals in this design doc the goal of not needing a second DS is achieved by treating LVMd as a library (I thought it was process) with the consequence of having two very different interface methods. My approach does result in retention of an independent process and grpc but means the interface between node and lvmd is the same rather it is deployed in the pod or deployed as a systemd service. The goals is I understand them are the same.  My approach has a slightly smaller code surface less to maintain and would make the embeded moot assuming I understood the goals correctly. While I am removing choices have I caused an unforeseen consequence by impacting a feature or function.

Reply:
Thanks for the back and forth I see the issue and will revise the approach to use embedded rather than side car